src/icp_prototype_backend/src/lib.rs

```diff
@@ -408,18 +450,21 @@ fn list_transactions(up_to_count: Option<u64>) -> Vec<Option<StoredTransactions>
     }
 
     let next_block = get_next_block();
-    let recent_block = next_block - 1;
```

Comment:
when simulated with boundary condition, assigning recent_block with next_block value would result unexpectedly
next_block = 102
recent_block = 102
up_to_count = 100
oldest_block = 1
tx_len = 101
if tx_len > 100 && 102 - 100 ≥  1 {
	start = 102 - 100 // this path is eval - 2
} else {
	start = oldest_block
}

Reply:
Yes, we need to figure this out.